### ISSUE TITLE
FGP_TV for Color images

### ISSUE THREAD

**Issue by epapoutsellis** [vais-ral/CCPi-Regularisation-Toolkit#150]
Hi @dkazanc,

I am trying to denoise color images, but I am getting a reconstruction which seems not to take the gradient across the Green channel. Is it tested for color images?

**Comment by dkazanc** [comment]
Hi, I haven't tested it for color images directly. Is that a N x M x 3 image you provide as an input? If yes, then it will be treated as a 3D image with a gradient calculated across color channels. I don't think it is desirable directly? 
I was thinking on incorporating an option where you specify how to treat the input data, e.g. treat 3D volume as a 2D one (slice by slice without correlation between) or 2D + channel (special smoothing across the channels). Do you reckon that would be useful?

**Comment by epapoutsellis** [comment]
No , I do not think we need it in CCPi. I was curious because I was doing some testing with colour image denoising with CIL. Also, in CIL, we have this kind of splitting as `correlation=Space` or `SpaceChannels`.

**Comment by dkazanc** [comment]
makes sense. this might be useful for 2D+time or 3D+time methods that I'd like to add at some point. May be then I could rethink the idea. The idea of being flexible on which norm is used with a chosen axis seems also interesting, so-called collaborative regularisation: https://arxiv.org/abs/1508.01308
